client/my-sites/checkout/cart/test/cart-coupon.js

```diff
@@ -176,10 +176,10 @@ describe( 'cart-coupon', () => {
 					} }
 				/>
 			);
-			applyCoupon.mockReset();
+			removeCoupon.mockReset();
 			component.find( '.cart__remove-link' ).simulate( 'click', event );
-			expect( applyCoupon.mock.calls.length ).toBe( 1 );
-			expect( applyCoupon.mock.calls[ 0 ][ 0 ] ).toBe( '' );
+			expect( removeCoupon.mock.calls.length ).toBe( 1 );
```

Comment:
You can use [`.toHaveBeenCalledTimes( 1 )`](https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalledtimesnumber)